rust/src/storage/mod.rs

```diff
@@ -180,6 +180,12 @@ pub enum StorageError {
         source: rusoto_core::RusotoError<rusoto_s3::PutObjectError>,
     },
     #[cfg(feature = "s3")]
+    #[error("Failed to delete S3 object: {0}")]
+    S3Delete(#[from] rusoto_core::RusotoError<rusoto_s3::DeleteObjectError>),
+    #[cfg(feature = "s3")]
+    #[error("Failed to copy S3 object: {0}")]
+    S3Copy(#[from] rusoto_core::RusotoError<rusoto_s3::CopyObjectError>),
+    #[cfg(feature = "s3")]
```

Comment:
It is different to the s3 errors above, I feel like with this approach it's cleaner looking and easier to pattern match, however it differs from other s3 errors, should I put it back, OR should I change other, existing errors